clients/pubsub-common/src/main/java/org/eclipse/hono/client/pubsub/PubSubMessageHelper.java

```diff
@@ -70,12 +73,78 @@ private static GoogleCredentials getCredentials() {
     /**
      * Gets the topic name with the given prefix.
      *
-     * @param topic The endpoint of the topic (e.g. event)
-     * @param prefix The prefix of the Pub/Sub topic, it's either the tenant ID or the adapter instance ID
+     * @param topic The endpoint of the topic (e.g. event).
+     * @param prefix The prefix of the Pub/Sub topic, it's either the tenant ID or the adapter instance ID.
      * @return The topic containing the prefix identifier and the endpoint.
      */
     public static String getTopicName(final String topic, final String prefix) {
-        return String.format("%s.%s", prefix, topic);
+        return getTopicName(topic, prefix, new ArrayList<>());
+    }
+
+    /**
+     * Gets the topic name with the given prefix and subtopics.
+     *
+     * @param topic The endpoint of the topic (e.g. event).
+     * @param prefix The prefix of the Pub/Sub topic, it's either the tenant ID or the adapter instance ID.
+     * @param subtopics A list with the subtopics.
+     * @return The topic containing the prefix identifier, the endpoint and all the subtopics.
+     */
+    public static String getTopicName(final String topic, final String prefix, final List<String> subtopics) {
+        final StringBuilder topicBuilder = new StringBuilder();
+        topicBuilder.append(prefix).append(".").append(topic);
+        for (String subtopic : subtopics) {
+            topicBuilder.append(".").append(subtopic);
+        }
+        return topicBuilder.toString();
+    }
+
+    /**
+     * Gets the subtopics from the message attributes.
+     *
+     * @param attributesMap The attribute map.
+     * @return An ordered list containing all the subtopics.
+     */
+    public static List<String> getSubtopics(final Map<String, String> attributesMap) {
+        String origAddress = getAttributesValue(attributesMap, MessageHelper.APP_PROPERTY_ORIG_ADDRESS)
+                .orElse("");
+        origAddress = origAddress.startsWith("/") ? origAddress.substring(1) : origAddress;
+        final List<String> origAddressSplit = new ArrayList<>(Arrays.stream(origAddress.split("/")).toList());
+        // Subtopics are located starting at the 4th position (e.g. event/tenantId/deviceId/subtopic1/subtopic2/...).
+        if (origAddressSplit.size() < 4) {
+            return new ArrayList<>();
+        }
```

Comment:
I assume that this means that subtopics can only be used when specifying a full address including endpoint, tenant and device ID, right?

Reply:
That is correct.